skill-tree /src/main/java/com/RDS/skilltree/utils/JWTUtils.java

```diff
@@ -0,0 +1,86 @@
+package com.RDS.skilltree.utils;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+@Slf4j
+public class JWTUtils {
+
+    @Value("${jwt.rds.public.key}")
+    private String publicRDSKeyString;
+    private static ConcurrentHashMap<String, RSAPublicKey> publicKeyCache = new ConcurrentHashMap<>();
```

Comment:
Since the class is a managed bean, you might want to avoid using the static modifier for the cache. Instead, keep it private and final.

Reply:
Its not being used, will remove this.

Reply:
Removed this